pkg/ovs/openflow/ofctrl_flow.go

```diff
@@ -134,11 +134,16 @@ func (f *ofFlow) CopyToBuilder(priority uint16) FlowBuilder {
 
 // ToBuilder returns a new FlowBuilder with all the contents of the original Flow.
 func (f *ofFlow) ToBuilder() FlowBuilder {
-	// TODO: use exported fields from ofFlow and remove nolint:govet
-	flow := *f.Flow //nolint:govet
+	flow := &ofctrl.Flow{
+		Table:      f.Flow.Table,
+		CookieID:   f.Flow.CookieID,
+		CookieMask: f.Flow.CookieMask,
+		Match:      f.Flow.Match,
+	}
+	f.Flow.CopyActionsToNewFlow(flow)
```

Comment:
I'm surprised we didn't go with something like `flow := f.Flow.Copy()` to take care of all the fields at once, but as long as it works it's good enough for me

Reply:
Rebased. In terms of the method signature I'll let @wenyingd comment since there are some fields in the flow that we probably don't want to get copied out (the mutex for sure :) but this could be tracked after v0.9.

Reply:
Right. I also remember it is due to we want not to copy some intermediate state in all cases.

Reply:
But here could you explain a little more the issue to fix? I especially want to understand whether only for part of cases we have the issue and so can not reuse the original flow directly, and what cases.

Believe originally we decided to reuse the flow in FlowBuilder() because we feel it is safe at that time, and only for some other cases we do CopyActionsToNewFlow().

Reply:
There are two use cases essentially when it comes to copying the flow to FlowBuilder. The first case is when the action of the flow needs to be updated, but we wish to keep the same flow match etc. In this case we construct a new ofctrl.Flow with the fields we care about, as defined in CopyToBuilder(). The second use case is when we simply want to update the priority of a flow. Before this PR, we used to build a new flow by copying the original flow struct and reset the priority. The old flow is deleted by calling del on the original flow, which is untouched. Now that  ofctrl.Flow in ofFlow is a pointer rather than a struct, it is no longer safe to do so. When resetting the priority for the new flow the priority of the original one gets modified at the same time, so the del operation will not work as expected.

Reply:
Thanks for the explanation. Then FlowBuilder() can call CopyToBuilder() to remove duplicated code.
Also I think maybe consider to rename these two funcs, like CopyToBuilder() and CopyToBuilderWithActions()?

Anyway, better to get review from @wenyingd as well.

Reply:
Merged ToBuilder and CopyToBuilder into one function to address the comment